packages/web-app-files/src/mixins/fileActions.js

```diff
@@ -260,30 +259,25 @@ export default {
           class: `oc-files-actions-${app.name}-trigger`,
           isEnabled: () => true,
           canBeDefault: defaultApplication === app.name,
-          handler: () => this.$_fileActions_openLink(app.name, fileId),
+          handler: () => this.$_fileActions_openLink(app.name, webDavPath),
           label: () => this.$gettextInterpolate(label, { appName: app.name })
         }
       })
     },
 
-    $_fileActions_openLink(appName, resourceId) {
+    $_fileActions_openLink(appName, filePath) {
       const routeOpts = this.$_fileActions__routeOpts(
         {
           routeName: 'external-apps'
         },
+        filePath,
         undefined,
-        resourceId,
         undefined
       )
 
-      routeOpts.params.appName = appName
-
       routeOpts.query = {
```

Comment:
If you only need to push the `app` var into the query, then `routeOpts.query.app = appName` does the same and only needs one line of code.